pkg/containerdutil/helpers.go

```diff
@@ -0,0 +1,43 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package containerdutil
+
+import (
+	"context"
+
+	"github.com/containerd/containerd/content"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type ReadBlob func(ctx context.Context, provider Provider, desc ocispec.Descriptor) ([]byte, error)
+
+func WithCache() ReadBlob {
```

Comment:
The name is not very clear. It's used as `containerdutil.WithCache` but not sure what the return can do. Suggest to call it `ReadBlobWithCache`.

Reply:
Agreed.
Will come with next push.